File: 01 - Dart/helloworld.dart
void main() {
  // Primer programa en Dart
  /*
  * Este programa imprime "Hello, World!" en la consola.
  */
  print("Hello, World!");

  // Declaración de variables
  var helloDart = "Hello, Dart!";
  // Se modifica el valor de la variable helloDart
  helloDart = "Hello, Dart Programming!";
  print(helloDart);

  var myInt = 7;
  var myDouble = 3.14;
  myInt = myInt + myDouble.toInt();
  myDouble = myDouble + myInt.toDouble();
  print(myInt);
  print(myDouble);

  var myBool = true;
  myBool = !myBool; // Cambia el valor de myBool a false
  print(myBool);

  // Uso de constantes
  final pi = 3.14159; // No se puede cambiar el valor de pi
  const gravity = 9.81; // Constante de gravedad
  print(pi);
  print(gravity);

  //La diferencia entre final y const es que final se puede asignar una vez y su valor se determina en tiempo de ejecución, mientras que const es una constante en tiempo de compilación y su valor debe ser conocido en el momento de la compilación.

  // Control de flujo
  if (myInt > 10) {
    print("myInt es mayor que 10");
  } else if (myInt == 10) {
    print("myInt es igual a 10");
  } else {
    print("myInt es menor que 10");
  }
  myFunction();
  String result = myFunctionWithReturn();
  print(result);

  var myList = [1, 2, 3, 4, 5];
  print(myList);
  print(myList[0]); // Imprime el primer elemento de la lista

  var mySet = {1, 2, 3, 4, 5, 1};
  print(mySet);
  // Un set no permite elementos duplicados, por lo que el valor 1 solo se almacena una vez

  var myMap = {"nombre": "Alejandro", "Edad": 30, "Ciudad": "Valencia"};
  print(myMap);
  print(myMap["nombre"]); // Imprime el valor asociado a la clave "nombre"

  var myMap2 = Map<String, int>();
  myMap2["uno"] = 1;
  myMap2["dos"] = 2;
  print(myMap2);
  // Un mapa es una colección de pares clave-valor, donde cada clave es única y se asocia a un valor. En este caso, el mapa myMap2 tiene claves de tipo String y valores de tipo int.

  for (final num in myList) {
    print(num);
  }

  var myCounter = 0;
  while (myCounter < myInt) {
    print("Contador: $myCounter");
    myCounter++;
  }

  var myClass = MyClass("Alejandro", 30);
  myClass.displayInfo();
  print(myClass.name); // Imprime el nombre de la instancia de MyClass

  print(MyEnum.dart); // Imprime el valor del enum MyEnum
}

void myFunction() {
  print("Se llama a una función");
}

String myFunctionWithReturn() {
  return "Esta función devuelve un string";
}

class MyClass {
  String name;
  int age;

  MyClass(this.name, this.age);

  void displayInfo() {
    print("Nombre: $name, Edad: $age");
  }
}

enum MyEnum { dart, python }

File: 02 - Python/helloworld.py
# Primer programa en Python
print("Hola, Python!")

'''
Este 
es
un 
comentario
'''
# Variables y tipos de datos
my_string = "Hola, mundo!"
print(my_string)
print(type(my_string))

my_string = 6
print(my_string)
print(type(my_string))

my_int = 10
my_int = my_int + 5
print(my_int)
print(my_int * 2)
print(my_int)

my_float = 3.14
print(my_float)
my_float = my_float * my_int
print(my_float)

my_bool = True
print(my_bool)
my_bool = not my_bool
print(my_bool)

# Concatenación de cadenas
print("El valor de my_string es: " + str(my_int))
print(f"El valor de my_string es: {my_int}")

# Listas y diccionarios
my_list = [my_int, my_float, my_bool, my_string]
print(type(my_list))
print(my_list)

my_dict = {
    "entero": my_int,
    "flotante": my_float,
    "booleano": my_bool,
    "cadena": my_string
}
print(type(my_dict))
print(my_dict)
print(my_dict["entero"])



# Set (conjunto de elementos únicos)
my_set = {my_int, my_float, my_bool, my_string}
print(type(my_set))
print(my_set)

# Tupla (inmutable)
my_tuple = (my_int, my_float, my_bool, my_string)
print(type(my_tuple))
print(my_tuple)

# Condicionales
if my_int > 10 and my_bool:
    print("my_int es mayor que 10")
elif my_int == 10 or my_bool:
    print("my_int es igual a 10")
else:
    print("my_int es menor que 10")

# Bucles
for my_item in my_list:
    print(my_item)

for i in range(5):
    print(i)

def my_function(x):
    return x * 2

result = my_function(2)
print(result)


class MyClass:
    def __init__(self, my_name):
        print("Se ha creado una instancia de MyClass")
        self.my_name = my_name
    def print_name(self):
        print(self.my_name)

my_class = MyClass("Alejandro")
my_class.my_name = "Juan"
my_class.print_name()